modules/nf-core/universc/main.nf

```diff
@@ -0,0 +1,117 @@
+process UNIVERSC {
+    tag "$meta.id"
+    label 'process_medium'
+
+    if (params.enable_conda) {
+        exit 1, "Conda environments cannot be used when using the Cell Ranger tool. Please use docker or singularity containers."
+        conda (params.enable_conda ? "hcc::cellranger=3.0.2" : null)
+    }
+    container "tomkellygenetics/universc:1.2.4"
+    containerOptions = "--user root"
+
+    input:
+    tuple val(meta), path(reads)
+    path  reference
+
+
+    output:
+    tuple val(meta), path("sample-${meta.id}/outs/*"), emit: outs
+    path "versions.yml"                              , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def sample_arg = meta.samples.unique().join(",")
+    def reference_name = reference.name
+    def input_reads = meta.single_end ? "--file $reads" : "-R1 ${reads[0]} -R2 ${reads[1]}"
+    """
+    bash /universc/launch_universc.sh \\
+        --id 'sample-${meta.id}' \\
+        ${input_reads} \\
+        --technology '${meta.technology}' \\
+        --chemistry '${meta.chemistry}' \\
+        --reference ${reference_name} \\
+        --description ${sample_arg} \\
+        --jobmode "local" \\
+        --localcores ${task.cpus} \\
+        --localmem ${task.memory.toGiga()} \\
+        --per-cell-data \\
+        $args
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        cellranger:  \$(echo \$(cellranger count --version 2>&1 | head -n 2 | tail -n 1 | sed 's/^.* //g' | sed 's/(//g' | sed 's/)//g' ))
+        universc:  \$(echo \$(bash /universc/launch_universc.sh --version | grep version | grep universc  | sed 's/^.* //g' ))
+    END_VERSIONS
+    """
+
+
+    stub:
+    """
+    mkdir -p "sample-${meta.id}/outs/"
+    touch sample-${meta.id}/outs/fake_file.txt
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        cellranger:  \$(echo \$(cellranger count --version 2>&1 | head -n 2 | tail -n 1 | sed 's/^.* //g' | sed 's/(//g' | sed 's/)//g' ))
+        universc:  \$(echo \$(bash /universc/launch_universc.sh --version | grep version | grep universc | sed 's/^.* //g' ))
+    END_VERSIONS
+    """
+}
+
+process UNIVERSC_CELLRANGER_OS_COUNT {
```

Comment:
What's the reasoning behind this not being its own module(in its own `main.nf`)?